### ISSUE TITLE
w/o & w  dcn, the network have different trained results

### ISSUE THREAD

**Issue by huapohen** [Epiphqny/VisTR#35]
when I drop the dcn at the last layer of fpn, the loss can not descend

**Comment by huapohen** [comment]
Thank you for sharing this perfect work!

**Comment by huapohen** [comment]
谢谢❀仙女这么棒的工作 ^__^

**Comment by huapohen** [comment]
self.class_embed = nn.Linear(hidden_dim, num_classes + 1)

**Comment by huapohen** [comment]
if  args.dataset_file == "ytvos":                                                         # line 313   in vistr.py
    num_classes = 41                                                                          # line 314

self.class_embed = nn.Linear(hidden_dim, num_classes + 1)           # line 39

target_classes = torch.full(src_logits.shape[:2], self.num_classes,     # line 126
loss_ce = F.cross_entropy(src_logits.transpose(1, 2), target_classes, self.empty_weight)    # line 130

I confused by the output dim of self.class_embed: 41+1=42, instead of 41?    # Queation about classes index #11

**Comment by huapohen** [comment]
I might understand
